packages/costa/src/runtime.ts

```diff
@@ -131,8 +131,10 @@ export class CostaRuntime {
       pkg = selectNpmPackage(meta, source);
     } else if (source.from === 'git') {
       debug(`requesting the url ${source.uri}...`);
-      const { hostname, pathname } = source.urlObject;
-      pkg = await fetchPackageJsonFromGit(`git@${hostname}:${pathname}`, 'HEAD');
+      const { hostname, auth, hash } = source.urlObject;
+      let pathname = source.urlObject.pathname.replace(/^\/:?/, '');
+      const remote = `${auth || 'git'}@${hostname}:${pathname}${hash || ''}`;
+      pkg = await fetchPackageJsonFromGit(remote, 'HEAD');
```

Comment:
This is merged by #238, you can rebase from `master`.

Reply:
has rebased